src/cli.py

```diff
@@ -19,25 +19,75 @@
 
 '''  to run > sudo LD_LIBRARY_PATH=../libcsp/build PYTHONPATH=../libcsp/build python3 src/cli.py -I uart -d /dev/ttyUSB1  '''
 import time
+import libcsp_py3 as libcsp
+import unicodedata
 from groundStation import groundStation
 
 opts = groundStation.options()
 gs = groundStation.groundStation(opts.getOptions())
 flag = groundStation.GracefulExiter()
 
 def cli():
+
+    sysVals = groundStation.SystemValues()
+
     while True:
         if flag.exit():
-            print('Exiting receiving loop')
+            print('Exiting receiving loop\n')
             flag.reset()
             return
         try:
             server, port, toSend = gs.getInput(prompt='to send: ')
-            if server == 24:
-                # This is a direct UART command to a ground station EnduroSat transceiver to enter PIPE
-                # Can be deleted for flight
-                gs.__setPIPE__()
+            data = bytearray(libcsp.packet_get_data(toSend))
+            cspData = bytearray(libcsp.packet_content(toSend))
+            print("toSend csp packet bytearray: ", cspData)
+
+            if (
+                server == sysVals.APP_DICT.get('OBC') and 
+                port == sysVals.SERVICES.get('SCHEDULER').get('port') and
+                data[0] == sysVals.SERVICES.get('SCHEDULER').get('subservice').get('SET_SCHEDULE').get('subPort')
+                ):
+
+                print('entered SET_SCHEDULE\n')
+
+                # open the scheduler text file as an array of strings
+                with open('schedule.txt') as f:
+                    cmdList = f.readlines()
+
+                #create an empty list, and create another list of csp objects
+                schedule = list()
+
+                # for each line of command, parse the packet
+                if len(cmdList) > 0:
+                    for i in range(0, len(cmdList)):
+
+                        # parse the time and command, then embed the command as a CSP packet
+                        schedulerObj = groundStation.embedCSP(cmdList[i])
+                        scheduler = schedulerObj.embedCSP()
+                        
+                        # append the list
+                        schedule.append(scheduler)
+                        print("list of schedules: ", schedule)
+                        scheduledTime = schedule[i]['time']
+                        scheduledDst = schedule[i]['dst']
+                        dst = (scheduledDst).to_bytes(2, byteorder='big')
+                        scheduledDport = schedule[i]['dport']
+                        dport = (scheduledDport).to_bytes(2, byteorder='big')
+                        scheduleSubservice = schedule[i]['subservice']
+                        returnPacket = bytearray(libcsp.packet_content(scheduleSubservice))
```

Comment:
This causes an exception because packet_content doesn't exist (yet). I think you should change this to packet_get_data and not bother adding another python binding.

Reply:
yup changed